src/osd/ECUtil.cc

```diff
@@ -69,27 +67,25 @@ int ECUtil::decode(
   }
 
   set<int> avail;
-  for (auto i = to_decode.begin();
-       i != to_decode.end();
-       ++i) {
-    assert(i->second.length() != 0);
-    avail.insert(i->first);
+  for (auto &&i : to_decode) {
+    assert(i.second.length() != 0);
+    avail.insert(i.first);
   }
 
   map<int, vector<pair<int, int>>> min;
-  assert(ec_impl->minimum_to_decode(need, avail, &min) == 0);
+  int r = ec_impl->minimum_to_decode(need, avail, &min) == 0;
```

Comment:
`r` is a boolean casted to int, and it should always be `1`.

Reply:
updated this!